split Zdivisibility from Znumtheory, deprecate most of Znumtheory

andres-erbsen opened the pull request:
Repeat of approved PR https://github.com/coq/coq/pull/19789

Overlays (optional since addition of a compat Require):

- [x] https://github.com/mit-plv/bedrock2/pull/455
- [x] https://github.com/mit-plv/fiat-crypto/pull/2081
- [x] https://github.com/AbsInt/CompCert/pull/548
- [ ] https://github.com/math-comp/mczify/pull/60
- [ ] https://gitlab.mpi-sws.org/iris/stdpp/-/merge_requests/605

proux01 commented:
According to the last message on the original PR https://github.com/rocq-prover/rocq/pull/19789#issuecomment-2586735608 I was just waiting for you to open the PR here (I even did the rebase for you, you just had to click the provided link) and promised to review/merge it quickly. Alas, this was four months ago, I no longer have time for it right now.
So if no one else takes over, feel free to merge if you're happy with it, once CI is green (it seems pretty unhappy right now).

andres-erbsen commented:
Note that this PR is not the straightforward rebase from january but rather a version adapted to work with the component-dependency-restricted build, including some changes to the component dependencies.

andres-erbsen commented:
I added a compat require of Znumtheory in ZArith, split up the changelog by category, and changed deprecations to say 9.1.

I intend to merge if CI passes.
